node/builder.go

```diff
@@ -272,28 +272,40 @@ func (nc *Builder) build(ctx context.Context) (*Node, error) {
 	outbox := message.NewOutbox(fcWallet, consensus.NewOutboundMessageValidator(), msgQueue, msgPublisher, outboxPolicy, chainStore, chainState)
 
 	nd := &Node{
-		blockservice: bservice,
-		Blockstore:   bs,
-		cborStore:    &ipldCborStore,
-		Clock:        nc.Clock,
-		Consensus:    nodeConsensus,
-		ChainReader:  chainStore,
-		ChainSynced:  moresync.NewLatch(1),
-		MessageStore: messageStore,
-		Syncer:       chainSyncer,
-		PowerTable:   powerTable,
-		PeerTracker:  peerTracker,
-		Fetcher:      fetcher,
-		Exchange:     bswap,
-		host:         peerHost,
-		Inbox:        inbox,
-		OfflineMode:  nc.OfflineMode,
-		Outbox:       outbox,
-		NetworkName:  network,
-		PeerHost:     peerHost,
-		Repo:         nc.Repo,
-		Wallet:       fcWallet,
-		Router:       router,
+		Clock:       nc.Clock,
+		OfflineMode: nc.OfflineMode,
+		Repo:        nc.Repo,
+		Network: NetworkSubmodule{
+			host:        peerHost,
+			PeerHost:    peerHost,
+			NetworkName: network,
+			PeerTracker: peerTracker,
+			Router:      router,
+		},
+		Wallet: WalletSubmodule{
+			Wallet: fcWallet,
+		},
+		Messaging: MessagingSubmodule{
+			Inbox:  inbox,
+			Outbox: outbox,
+		},
+		Blockstore: BlockstoreSubmodule{
+			blockservice: bservice,
+			Blockstore:   bs,
+			cborStore:    &ipldCborStore,
+		},
+		StorageNetworking: StorageNetworkingSubmodule{
+			Exchange: bswap,
+		},
+		Chain: ChainSubmodule{
+			Fetcher:      fetcher,
+			Consensus:    nodeConsensus,
+			ChainReader:  chainStore,
+			ChainSynced:  moresync.NewLatch(1),
+			MessageStore: messageStore,
+			Syncer:       chainSyncer,
+			PowerTable:   powerTable,
+		},
```

Comment:
`BlockMiningSubmodule` is missing here. I see that, unlike other submodules, it's not instantiated until started. That at least deserves a clear comment here. I hope that future work will resolve that so that it can be instantiated without starting and we can avoid the nil references.

Reply:
Correct. The next PR add the call to initialize all the nil submodules, and attempts to move some of the initialization code back here. It might take a few PR's to fully separate initialization from start.